calico/felix/test/test_endpoint.py

```diff
@@ -53,18 +53,21 @@ def test_remove_deleted(self):
         """
         ep = endpoint.Endpoint(str(uuid.uuid4()), 'aa:bb:cc:dd:ee:ff')
 
-        with mock.patch('calico.felix.devices.tap_exists', \
-                        side_effect=[True, False]) as mock_exists,\
-             mock.patch('calico.felix.devices.list_tap_ips', \
-                         return_value = set(["1.2.3.4", "1.2.3.5"])) as mock_list, \
-             mock.patch('calico.felix.devices.del_route', \
-                        side_effect=futils.FailedSystemCall("blah",
-                                                            ["dummy", "args"],
-                                                            1,
-                                                            "",
-                                                            "")) as mock_del_route, \
-             mock.patch('calico.felix.frules.del_rules') as mock_del_rules:
-            ep.remove(iptables_state)
+        with mock.patch('calico.felix.devices.tap_exists',
+                        side_effect=[True, False]) as mock_exists:
+            with mock.patch(
+                'calico.felix.devices.list_tap_ips',
+                return_value = set(["1.2.3.4", "1.2.3.5"])) as mock_list:
+                with mock.patch(
+                    'calico.felix.devices.del_route',
+                    side_effect=futils.FailedSystemCall("blah",
+                                                        ["dummy", "args"],
+                                                        1,
+                                                        "",
+                                                        "")) as mock_del_route:
+                    with mock.patch(
+                        'calico.felix.frules.del_rules') as mock_del_rules:
+                        ep.remove(iptables_state)
```

Comment:
Oh yeah! I knew that was somewhere, but I could only see contextlib.ExitStack in recent docs so I got tunnel vision.